### ISSUE TITLE
Unsetting all styles, also known as Clear formatting

### ISSUE THREAD

**Issue by midamon** [memspace/zefyr#441]
Greetings,
Thank you for this cool package.

I was wondering how would I go about adding support for clear formatting (unsetting all NotusStyles) in a selected range?
It seems that it would be proper to add a heuristic rule just for that, but would that be a new special type of rule or would it be a catch-all format rule?

**Comment by cgestes** [comment]
Hi, I am not sure I understand how you would like to achieve it.

I think you should implement a Delta transformation, removing all the formating in the appropriate way and apply that transformation once you click on the button. 

You need confirmation from @pulyaevskiy but it seems it could be a function on NotusDocument: ```document.clearFormatting(range..)```

**Comment by pulyaevskiy** [comment]
Yeah, heuristic rule probably won't be of much help here, at least in the way such rules are used today. Heuristic rules can only act on individual insert, delete or format (single retain operation in Delta)  actions triggered by a user. Even though "clearFormatting" is a kind of "format" action, it's a complex one which cannot be described as a single Delta operation.

`NotusDocument.clearFormatting(int index, int length)` sounds like a good option. Its implementation would likely be similar to how heuristic rules work though.

1. Get the document Delta and create DeltaIterator.
2. Skip everything before `index`
3. Continue iterating until reached `length` and for each operation produce equivalent of `unset` attributes for the op
4. Add all unset styles to a new delta using `retain` operation.
5. After reached `length`, continue iterating until reached first newline character, unset any styles on it and add to the resulting delta (this is to make sure that if a line is only selected partially and has block style attributes (quote, list) those are cleared as well).
6. Call `compose(...)` to apply the result to a document.
